spectrum.js

```diff
@@ -324,6 +327,79 @@
             // Prevent clicks from bubbling up to document.  This would cause it to be hidden.
             container.click(stopPropagation);
 
+            // Handle arrow keys and Enter -- for keyboard navigation
+            paletteContainer.keydown(function(e) {
+
+                var focusedColor;
+
+                // Arrow key
+                if ($.inArray(e.keyCode, [37, 38, 39, 40]) >= 0) {
+
+                    // Currently-focused color
+                    focusedColor = $(this).find(".sp-thumb-focus");
+
+                    var newFocusedColor;
+                  
+                    if (focusedColor.length > 0) {
+                        // Find the color above/below/before/after the currently-focused color
+                        if (e.keyCode == 37 || e.keyCode == 39) {
+                            // left or right
+                            newFocusedColor = e.keyCode == 37 ? focusedColor.prev() : focusedColor.next();
+                        }
+                        else if (e.keyCode == 38 || e.keyCode == 40) {
+                            // up or down
+                            var row = e.keyCode == 38 ? focusedColor.parent().prev() : focusedColor.parent().next();
+                            if (row.length > 0) {
+                                newFocusedColor = row.children().eq(focusedColor.index());
+                            }
+                        }
+                    }
+                    else {
+                        // No currently-focused color, so just choose the first one
+                        newFocusedColor = $(this).find(".sp-thumb-el").first();
+                    }
+ 
+                    // Give focus to the new focused color
+                    if (newFocusedColor && newFocusedColor.length > 0)
+                    {
+                        newFocusedColor.focus();
+
+                        // focusedColor.attr("tabindex", -1);
+                        focusedColor.removeClass("sp-thumb-focus");
+
+                        // newFocusedColor.attr("tabindex", 0);
+                        newFocusedColor.addClass("sp-thumb-focus");
+                    }
+
+                    return false;
+                }
+
+                // Enter key
+                else if (e.keyCode == 13) {
+
+                    // Currently-focused color
+                    focusedColor = $(this).find(".sp-thumb-focus");
+                    if (focusedColor) {
+                        set(focusedColor.data("color"));
+                        move();
+                        updateOriginalInput(true);
+                        if (opts.hideAfterPaletteSelect) {
+                            hide();
+                        }
+                    }
+
+                    return false;
+                }
+
+                // Escape (TODO)
+                else if (e.keyCode == 27) {
+
+                    // TODO: close()
```

Comment:
Yes, this can be removed also, since you implemented the Escape behavior earlier . Done.